feat(flags): Port `local_evaluation` endpoint to Rust

haacked opened the pull request:
## Problem

Most requests to `/local_evaluation` should be completely resolved from the cache. We already return flag definitions from the cache. We plan to use HyperCache to handle authentication as well.

But before we do that, we want a lightweight service to serve up local_evaluation. This PR is step 1 in that.

## Changes

Adds a new `/flags/definitions` endpoint to serve up feature flag definitions used by SDK clients to implement local evaluation of feature flags.

This leverages the `HyperCache` to return flag definitions. This does not use `HyperCache` for authentication yet. That'll come in a follow-up.

## How did you test this code?

Unit Tests
Manual Tests

__No Access__
- [x] Invalid PAT
- [x] Valid PAT, but wrong org scope
- [x] Valid PAT, but wrong team scope
- [x] Valid PAT, but wrong feature scope
- [x] Invalid secret api token
- [x] Valid secret api token, but wrong team
- [x] Valid PAT, all access, but user not in org
- [x] Invalid project api key (?token)

__Successful Access__
- [x] Valid PAT - org: All Access, scopes: All Access (default org)
- [x] Valid PAT - org: All Access, scopes: All Access (another org the user has access to)
- [x] Valid PAT - project: All Access, scopes: All Access (default project)
- [x] Valid PAT - project: All Access, scopes: All Access (another project the user has access to)
- [x] Valid PAT, specific teams:
- [x] Valid PAT, scoped team, proper scope
- [x] Valid PAT, scoped org, proper feature scope `feature_flag:read`
- [x] Valid PAT, scoped org, proper feature scope `feature_flag:write`
- [x] Valid secret api token
- [x] Valid backup secret api token
- [x] Cache miss correct response

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Important Context

1. This introduces a new endpoint, but nobody is calling it yet, so it's relatively safe in that regard.
2. It does not handle cache misses yet. That'll come in a follow-up.
3. It does not use Hypercache for auth yet. That's coming later.

haacked committed:
feat(flags): Add `local_evaluation` endpoint

This leverages the `HyperCache` to return flag definitions.

This does not use `HyperCache` for authentication yet. That'll come later.

gustavohstrassburger commented on rust/feature-flags/src/team/team_operations.rs:
```diff
@@ -84,48 +171,69 @@ impl Team {
         Ok(row)
     }
+
+    pub async fn from_pg_by_id(client: PostgresReader, team_id: i32) -> Result<Team, FlagError> {
```
Method not used, is it necessary?

gustavohstrassburger commented on rust/feature-flags/src/api/auth.rs:
```diff
@@ -0,0 +1,385 @@
+                row.try_get("scoped_organizations").ok();
+
+            // Fetch the team's organization_id
+            let (_, team_organization_id) =
```
Do we really need to fetch the Team again? It was already fetched earlier and received via param.

haacked replied:
> isn't it better to also get that column earlier

Yeah, but because existing cache entries wouldn't have this value, I didn't want to deal with it now. But on thinking about it, I'll add it to the `Team` struct as an optional value and if it's missing, we can fetch just that value.

gustavohstrassburger reviewed (approved):
Solid PR, nice work.

I've left a few comments and also noticed a couple of features that are missing from the port, just want to know that you are aware of those.

Since this endpoint is not being used yet, don't feel blocked by those.

Features not ported:
- Throttle control (`LocalEvaluationThrottle`)
- Quota (`DECIDE_FEATURE_FLAG_QUOTA_CHECK`)
- Metrics (`LOCAL_EVALUATION_REQUEST_COUNTER`)
- Not incrementing the request count (`increment_request_count`)
- Auth by temporary token ("temporary_token")

haacked committed:
Rename function for clarity

Rename fetch_team_with_redis_fallback to fetch_team_from_redis_with_fallback
to make it clearer that we're fetching from Redis with a fallback to PostgreSQL,
not using Redis as the fallback.

haacked committed:
Add organization_id to Team struct

And update code so we always query this. When retrieving from cache, if it's not there, we do a targeted query.

haacked committed:
Treat teams without organization_id as cache miss

When a team is fetched from Redis without organization_id (old cache entry),
treat it as a cache miss. This triggers a database fetch and cache update,
ensuring all callers automatically get complete teams with organization_id.

This eliminates the need for fallback logic in validation functions and
provides automatic migration of old cache entries.
